File: 프로그래머스/unrated/120808. 분수의 덧셈/분수의 덧셈.py
def solution(numer1, denom1, numer2, denom2):
    answer = []
    if denom1 != denom2:
        denom = denom1
        numer1 *= denom2
        denom1 *= denom2
        numer2 *= denom
        denom2 *= denom
    resultNumer = numer1 + numer2
    resultDenom = denom1
    
    n = min(resultNumer, resultDenom)
    for i in range(n,1, -1):
        if resultNumer % i == 0 and resultDenom % i == 0:
            resultNumer = resultNumer // i
            resultDenom = resultDenom // i
    
    answer.append(resultNumer)
    answer.append(resultDenom)
    
    
    return answer

File: 프로그래머스/unrated/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int[] answer = new int[2];
        if (denom1 == denom2){
            answer[0] = (numer1+numer2);
            answer[1] = (denom1);
        }
        else{
            answer[0] = (numer1*denom2 + numer2*denom1);
            answer[1] = (denom1*denom2);
        }
        int i = answer[0];
        while(i>0){
            if(answer[0]%i == 0 && answer[1]%i == 0){
                answer[0] = answer[0]/i;
                answer[1] = answer[1]/i;
            }
            i -= 1;
        }
        
        return answer;
    }
}